### PR TITLE
Fortmatic

### PR DESCRIPTION
**Pull request by JamesDalboth**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

Essentially this PR makes use of the fortmatic provider.

Some notes:

1. If you have metamask being injected it will use this instead. This is intentional, if you want to use metamask we will let you use metamask, if you don't have metamask, don't worry, we will give you fortmatic
2. Formatic is slow. Not much we can do about this, but it is slower compared to metamask.It isn't too bad but if you click provenance of a piece it will take about 5 seconds for things to load in. Again we can always switch to metamask if we have it installed
3. Fortmatic does not like truffle contracts. Not sure why, but it doesn't, simple fix to change all contracts to a simple web3 contract. 
4. Fortmatic only works on test nets. This means if you run the app locally and you want to use fortmatic it will actually connect to the rinkeby network. This is actually fine because I have changed the contract logic to search for the contracts so we no longer require truffle to know where the contracts are stored

### Checklist
* [x] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [x] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [x] Will you squash when you merge this PR?

### CODE REVIEW

File: app/src/components/App.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ const ContractProvider: React.FC = ({ children }) => {
 
 const App: React.FC<AppProps> = (props) => {
   return (
-    <Web3Provider networkId={props.networkId} accounts={props.accounts}>
+    <Web3Provider web3={props.web3} networkId={props.networkId} accounts={props.accounts}>
```
Comment (DJRHails) [inline app/src/components/App.tsx]: Why are we injecting web3 in here as opposed to creating it inside?

**Review by DJRHails — CHANGES_REQUESTED** [review]
Only thing is I'm unsure why we're injecting from index / app as opposed to keeping it isolated to web3 specific files

File: app/src/components/App.tsx [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
-      <ContractProvider>
-        <SessionProvider address={props.accounts[0]} contracts={props.contracts}>
-          <Router {...props}/>
-        </SessionProvider>
-      </ContractProvider>
+    <Web3Provider>
+      <NameServiceProvider>
+        <ContractProvider>
+          <ArtistProvider>
+            <RegistryProvider>
```
Comment (DJRHails) [inline app/src/components/App.tsx:18]: Registry is a Contract?

File: app/src/components/Router.tsx [hunk 3]
```diff
@@ -2,15 +2,18 @@ import * as React from 'react';
...
 
 import * as View from '../views';
 
-import { AppProps } from './App';
 import Main from './Main';
 import NetworkAside from './NetworkAside';
 
-const Router: React.FC<AppProps> = ({ web3, accounts, contracts }) => {
+const Router: React.FC<{}> = () => {
   const { user } = useSessionContext();
```
Comment (DJRHails) [inline app/src/components/Router.tsx:14]: Find and replace React.FC<{}> with React.FC

File: app/src/providers/ContractProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }, [addressFromName, web3.eth.Contract]);
+
+  if (!contracts) {
+    return <Loading/>;
+  }
+
+  console.log('Contracts provided');
+
+  return (
+    <ContractContext.Provider value={{ contracts: contracts }}>
```
Comment (DJRHails) [inline app/src/providers/ContractProvider.tsx:56]: Consider expanding here so provider has value: Governance: ..., ArtistApplication: ....

Comment (DJRHails) [reply app/src/providers/ContractProvider.tsx:56]: As opposed to contracts:{Governance: ... Etc

**Review by DJRHails — CHANGES_REQUESTED** [review]

File: app/src/components/App.tsx (hunk 2 above)
Comment (JamesDalboth) [reply app/src/components/App.tsx:18]: Yeah this was something that was done before this PR. For somereason there was a RegistryProvider which just connects you to the contract even though we were also grabbing all contracts in index and passing those down. I imagine the provider option was to reduce the passing down of contracts. I agree something needs to be fixed and am happy to do that but would rather do in a seperate PR?

File: app/src/providers/ContractProvider.tsx (hunk 4 above)
Comment (JamesDalboth) [reply app/src/providers/ContractProvider.tsx:56]: Agree with the proposed change but would rather do with another PR to clean up contract stuff

**Review by DJRHails — APPROVED** [review]

**Pull request merged**